Fix migrations and preserved User table and Tasks features

KellyAH opened the pull request:
# Problem

Running migrations on the latest main branc is currently broken and throws an error when running `rails db:migrate`.

```
get-things-done-app/db/migrate/20201024193303_create_thoughts.rb:5: syntax error, unexpected tSYMBEG, expecting end t.has_one :user_category ^ 
get-things-done-app/db/migrate/20201024193303_create_thoughts.rb:13: syntax error, unexpected end, expecting end-of-input
```

# Cause
Lailson says as a Rails best practice, **do NOT edit old migration scripts**. Just make new migration scripts that makes the changes you want. Migrations broke because db/migrate/20201024193303_create_thoughts.rb was edited in commit https://github.com/KellyAH/get-things-done-app/commit/7c9d3fbe6b0897e07eac6f2a45a53bee6df89db3. Which threw the above error. 

And other errors were thrown complaining about priority failing to be created because it already existed when the migration scripts (which I deleted in this commit https://github.com/KellyAH/get-things-done-app/pull/9/commits/86b5f0b22ca75704eec22a6bcdb8472f52dc3d6d) were executed. They failed because they were trying to add a priority column to the thoughts table when it was already added by `20201024193303_create_thoughts.rb`. 

# Reference
[Rails Active Record Migrations guide](http://guides.rubyonrails.org/migrations.html)
https://stackify.com/rails-migration-a-complete-guide/

# Code Changes
This branch was cut off of `practice-rails-stuff` branch (see code changes in https://github.com/KellyAH/get-things-done-app/pull/6) which also had the latest main merged into it. So besides migration fixes and a tiny alert message feature, this branch **also contains**:
- all the features https://github.com/KellyAH/get-things-done-app/pull/6 had 
- commit https://github.com/KellyAH/get-things-done-app/commit/7c9d3fbe6b0897e07eac6f2a45a53bee6df89db3 which was merged to main Nov 27th.

**Structural and Behavioral**

**https://github.com/KellyAH/get-things-done-app/pull/6 Features**
- Make submit from on home page create new Thoughts
- Create `Tasks` data and display it on the homepage
- add the ability to destroy and edit `Tasks`
- Add ability to turn `Thoughts` into a `Tasks`

**Fixed migration scripts**
* Fixed the broken db/migrate/20201024193303_create_thoughts.rb and added ability to create a new thought when a user object doesn't exist. Because the app currentlyy doesn't have the capability to select a user on the Frontend. And creating a new thought was breaking without my change.  (see https://github.com/KellyAH/get-things-done-app/pull/9/commits/75e6d88f1d8cdc823a5e89364b72ecd60cc44971)
* deleted no longer needed migration scripts (see https://github.com/KellyAH/get-things-done-app/pull/9/commits/86b5f0b22ca75704eec22a6bcdb8472f52dc3d6d)

**New Feature**
* Added a red error message that displays the Thought object error when the app fails to create a Thought on the home page.

<img width="536" alt="red error message" src="https://user-images.githubusercontent.com/16140873/103397918-31ea2600-4aef-11eb-832f-c72e5209980a.png">


# Tests
Rebuilt DB successfully on my local `rails db:drop db:create db:migrate` and all main features to create tasks, thoughts etc work when manually testing the app on my local.

<img width="568" alt="app still works" src="https://user-images.githubusercontent.com/16140873/103397934-3ca4bb00-4aef-11eb-825b-4e613307bf88.png">

KellyAH committed:
save point

KellyAH committed:
display tasks on index

KellyAH committed:
add a link to index for coverting a thought into a task

KellyAH committed:
pre populate new task form with thought description

KellyAH committed:
try to make index submit button create a new thought

KellyAH committed:
turn thoughts into tasks

KellyAH committed:
fix form on index to create new thoughts

KellyAH committed:
add ability to destroy tasks

KellyAH committed:
add return to homepg to thoughts pages

KellyAH committed:
add ability to edit a task

KellyAH committed:
add comments

KellyAH committed:
chore: resolve merge conflicts

KellyAH committed:
refactor tasks show and edit using set_task

KellyAH committed:
add indent todos

KellyAH committed:
delete unneeded migrations

KellyAH committed:
fix migration scripts and create new thought

KellyAH committed:
display red alert message on homepage when create thought fails

KellyAH commented on app/controllers/tasks_controller.rb:
```diff
@@ -0,0 +1,73 @@
+  before_action :set_task, only: [:show, :edit, :update, :destroy]
+
+
+  def new
+    @task = Task.new(priority: "medium")
+    if params[:thought_id]
+      thought = Thought.find(params[:thought_id])
+      @task.description = thought.description
+    end
+  end
```
change to 

```suggestion
  def KDFHKHKDFKDF
    @task = Task.new(priority: "medium")
    if params[:thought_id]
      thought = Thought.find(params[:thought_id])
      @task.description = thought.description
    end
  end
```

KellyAH commented on app/views/homepage/index.html.erb:
```diff
@@ -1,14 +1,12 @@
-  <%= submit_tag("Submit") %>
+<%= form_with model: @new_thought do |form| %>
+  <%= form.text_field :description %>
+  <%= form.button "Create" %>
```
dsadsdsas

naomiquinones commented on app/views/homepage/index.html.erb:
```diff
@@ -1,14 +1,12 @@
+<p id="notice"><%= notice %></p>
+<p id="alert"><%= alert %></p>
+
 <h1>Welcome to The Get Things Done App.</h1>
```
```suggestion
<h1>Welcome to The Get Things Done App.</h1>

<p id="notice"><%= notice %></p>
<p id="alert"><%= alert %></p>

```

NOI-lgtm commented on app/controllers/thoughts_controller.rb:
```diff
@@ -12,11 +12,6 @@ def index
-  end
-
   # GET /thoughts/1/edit
   def edit
```
This is empty method

karolina-benitez reviewed (commented):
LGTM!

karolina-benitez reviewed (changes requested):
BLAH BLAH

naomiquinones commented on app/assets/stylesheets/scaffolds.scss:
```diff
@@ -36,6 +36,9 @@ div.actions {
 #notice {
   color: green; }
 
+#alert {
+  color: red; }
```
```suggestion
#alert {
  color: #cc1234;
  font-weight: bold; }
```

NOI-lgtm commented on app/controllers/thoughts_controller.rb:
```diff
@@ -12,11 +12,6 @@ def index
   def show
```
```suggestion
  # Delegating to find tasks id into private methods set_thought
  def show
```

naomiquinones reviewed (changes requested):
I added css

KellyAH committed:
Move alerts and notices on home page to below main title

We felt it looked better below the main title.

Co-authored-by: Naomi Quinones <52065567+naomiquinones@users.noreply.github.com>

KellyAH committed:
Bold alert text and make darker red color

Co-authored-by: Naomi Quinones <52065567+naomiquinones@users.noreply.github.com>

NOI-lgtm reviewed (commented):
Add comment in empty method

NOI-lgtm committed:
Added comments to empty method

It shows that the empty method is intentional and it's being called by the private method set_thought.

NOI-lgtm commented:
LGTM! 👍

NOI-lgtm reviewed (approved):
Amazing 💯 👍

karolina-benitez reviewed (approved):
👸🏻
